SwanCustomEnvironments/swancustomenvironments/scripts/builders/venv.sh

```diff
@@ -1,16 +1,36 @@
 #!/bin/bash
 
-# Create the environment
-python3 -m venv ${ENV_PATH} 2>&1
+# Create a middle layer for installing ipykernel, putting it apart from the user environment
+uv venv $SWAN_ENV --seed 2>&1
+source $SWAN_ENV/bin/activate
+uv pip install "ipykernel==${IPYKERNEL_VERSION}"
+SWAN_PACKAGES_PATH=$(python3 -c 'import sysconfig; print(sysconfig.get_paths()["purelib"])')
+deactivate
+
+if [ "${RESOLVED_REQ}" = true ]; then
+    uv venv ${ENV_PATH} --seed 2>&1
+else
+    python -m venv ${ENV_PATH} 2>&1
+fi
 
 # Activate the environment
 _log "Setting up the environment..."
 ACTIVATE_ENV_CMD="source ${ENV_PATH}/bin/activate"
 eval "${ACTIVATE_ENV_CMD}"
 
-# Install packages in the environment and the same ipykernel that the Jupyter server uses
+# Install user-requested packages in the environment
+# Use uv for better performance of resolved
```

Comment:
Nono, I was finishing my comment!

Reply:
```
# Install user-requested packages in the environment.
# Use uv for better performance if environment is fully resolved;
# Otherwise, use pip for resolution (more reliable long-term).
```